OpenRA.Mods.Common/HitShapes/Rectangle.cs

```diff
@@ -68,13 +86,24 @@ public WDist DistanceFromEdge(WVec v)
 
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
-			return DistanceFromEdge((pos - actor.CenterPosition).Rotate(-actor.Orientation));
+			var actorTargPos = actor.TargetingCenterPosition;
+
+			if (pos.Z <= actorTargPos.Z + VerticalTopOffset && pos.Z >= actorTargPos.Z + VerticalBottomOffset)
+				return DistanceFromEdge((pos - new WPos(actorTargPos.X, actorTargPos.Y, pos.Z)).Rotate(-actor.Orientation));
+			if (pos.Z > actorTargPos.Z + VerticalTopOffset)
+				return DistanceFromEdge((pos - (actorTargPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-actor.Orientation));
+			if (pos.Z < actorTargPos.Z + VerticalBottomOffset)
+				return DistanceFromEdge((pos - (actorTargPos + new WVec(0, 0, VerticalBottomOffset))).Rotate(-actor.Orientation));
+
+			return DistanceFromEdge((pos - actorTargPos).Rotate(-actor.Orientation));
 		}
 
 		public void DrawCombatOverlay(WorldRenderer wr, RgbaColorRenderer wcr, Actor actor)
 		{
-			var verts = combatOverlayVerts.Select(v => wr.ScreenPosition(actor.CenterPosition + v.Rotate(actor.Orientation)));
-			wcr.DrawPolygon(verts.ToArray(), 1, Color.Yellow);
+			var vertsTo = combatOverlayVertsTop.Select(v => wr.ScreenPosition(actor.TargetingCenterPosition + v.Rotate(actor.Orientation)));
```

Comment:
One more letter makes this a bit easier to understand at a glance.
Might as well change the below to `vertsBelow`, also.